modules/social_features/social_event/modules/social_event_invite/src/Form/EnrollInviteEmailForm.php

```diff
@@ -178,6 +200,45 @@ public function buildForm(array $form, FormStateInterface $form_state) {
       '#limit_validation_errors' => [],
     ];
 
+    // We should prevent invite enrollments if social_event_max_enroll is
+    // enabled and there are no left spots.
+    if ($this->moduleHandler->moduleExists('social_event_max_enroll')) {
+      $event_max_enroll_service = $this->eventMaxEnrollService;
+
+      $nid = $this->routeMatch->getRawParameter('node');
```

Comment:
Line 137 of this function is already getting the current node parameter from this route. I would suggest to store that on a variable, and use it at this part of the code. The function `getParameter` will always return the processed parameter value, in this case, will already be a node object, if it exists.

This means that, from lines 208 to 224 you don't need all those defensive checks. You can still check if the `$node` is a `nodeInstance` ( line 255 ) in case someone tries to access some invalid node.

Reply:
I have changed it.